File: tests/samples/java/Vulnerable.java
import java.sql.*;
import java.security.MessageDigest;
import java.io.*;
import java.net.URL;

public class Vulnerable {
    private static final String API_KEY = "sk-1234567890abcdef1234567890abcdef";
    private static final String DB_PASS = "admin123";

    public String getUser(String userId) {
        String query = "SELECT * FROM users WHERE id = '" + userId + "'";
        Statement stmt = connection.createStatement();
        return stmt.executeQuery(query).toString();
    }

    public void renderProfile(String userName) {
        String html = "<div>Welcome " + userName + "</div>";
        response.getWriter().write(html);
    }

    public void runCommand(String host) {
        Runtime.getRuntime().exec("ping -c 1 " + host);
    }

    public byte[] loadFile(String filename) {
        return new File("/var/uploads/" + filename).readAllBytes();
    }

    public Object restoreState(byte[] data) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(data));
        return ois.readObject();
    }

    public void executeCode(String code) throws Exception {
        ScriptEngine engine = new ScriptEngineManager().getEngineByName("JavaScript");
        engine.eval(code);
    }

    public void fetchRemote(String url) throws Exception {
        URL target = new URL(url);
        target.openConnection().getInputStream();
    }

    public String hashPassword(String pass) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");
        byte[] digest = md.digest(pass.getBytes());
        return bytesToHex(digest);
    }
}

File: tests/samples/js/vulnerable.js
const crypto = require("crypto");
const express = require("express");
const app = express();

const API_SECRET = "super-secret-token-1234567890";
const DB_CONNECTION = "mongodb://admin:password123@localhost:27017/mydb";

app.get("/user", (req, res) => {
  const userId = req.query.id;
  const query = "SELECT * FROM users WHERE id = '" + userId + "'";
  db.query(query);
  res.send("<div>Hello " + userId + "</div>");
});

app.get("/login", (req, res) => {
  const user = req.query.user;
  const pass = req.query.pass;
  const html = '<div>Welcome <b>' + user + '</b></div>';
  document.getElementById("content").innerHTML = html;
});

app.get("/profile", (req, res) => {
  const code = req.query.code;
  eval("var user = " + code + ";");
  res.json(user);
});

app.get("/admin", (req, res) => {
  const host = req.query.host;
  const child = require("child_process").exec("ping -c 1 " + host);
});

function hashPassword(password) {
  return crypto.createHash("md5").update(password).digest("hex");
}

const badFetch = (req) => {
  const url = req.query.url;
  return fetch(url);
};
